include/device.h

```diff
@@ -30,6 +30,39 @@ struct driver {
 
 typedef enum { DEV_BUS_NONE, DEV_BUS_PCI, DEV_BUS_ISA } device_bus_t;
 
+#define DEV_ATTR_MAX 16 /* maximum number of device attributes */
+
+typedef enum {
+  DA_NULL = 0,
+  /* FDT attributes */
+  DA_FDT_COMPATIBLE,
+  DA_FDT_PATH,
+  /* resource types */
+  DA_IOPORT,
+  DA_MEMORY,
+  DA_IRQ,
+  /* PCI bus identifiers */
+  DA_PCI_CLASSCODE,
+  DA_PCI_DEVICEID,
+  DA_PCI_VENDORID,
+} __packed da_tag_t;
```

Comment:
I'm not sure about this one, but is it possible that we may encounter unaligned access when using `__packed`?

Reply:
Packed unions will take the least space that can accommodate all members => in this case `char` will do. Normally it'd consume space for `int`.